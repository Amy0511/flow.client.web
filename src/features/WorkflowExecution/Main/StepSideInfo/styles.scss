@import "~@boomerang/boomerang-styles/lib/colors";

.c-step-side-info {
  position: absolute;
  right: 1rem;
  top: 22rem;
  padding: 1rem;
  background-color: $white;
  width: 25vw;
}

.s-step-side-info-title {
  font-size: 1.25rem;
  font-weight: 700;
  margin-left: 1rem;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.c-step-side-info-details {
  padding: 2rem 0 0 1.75rem;
  width: 100%;
}

.b-step-side-info-field {
  display: flex;
  justify-content: space-between;
  margin-bottom: 2rem;
}

.b-step-side-info-field__key {
  width: 30%;
  font-size: 1rem;
  font-weight: 600;
}

.b-step-side-info-field__value {
  width: 60%;
  font-size: 1rem;
  font-weight: 500;
  text-align: left;
  display: flex;
  align-items: center;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;

  img {
    margin-left: 1rem;
    height: 1rem;
    width: 1rem;
  }
}
