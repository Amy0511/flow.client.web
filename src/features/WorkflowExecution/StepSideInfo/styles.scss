@import "~@boomerang/boomerang-styles/lib/colors";

.c-step-side-info {
  width: 30%;
  padding: 2rem;
  background-color: $white;
  margin-top: 1rem;
  align-self: flex-end;
}

.s-step-side-info-title {
  font-size: 1.25rem;
  font-weight: 700;
}

.c-step-side-info-details {
  padding: 2rem 0 0 1.75rem;
  width: 100%;
}

.b-step-side-info-field {
  display: flex;
  justify-content: space-between;
  margin-bottom: 2rem;
}

.b-step-side-info-field__key {
  width: 30%;
  font-size: 1rem;
  font-weight: 600;
}

.b-step-side-info-field__value {
  width: 60%;
  font-size: 1rem;
  font-weight: 500;
  text-align: left;
}
