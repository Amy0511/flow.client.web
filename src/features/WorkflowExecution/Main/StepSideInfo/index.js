import React from "react";
import PropTypes from "prop-types";
import moment from "moment";
import getHumanizedDuration from "@boomerang/boomerang-utilities/lib/getHumanizedDuration";
import { ACTIVITY_STATUSES_TO_TEXT, ACTIVITY_STATUSES_TO_ICON } from "Constants/activityStatuses";
import "./styles.scss";

const StepSideInfo = ({ step }) => {
  return (
    <div className="c-step-side-info">
      <div className="s-step-side-info-title">Summary</div>
      <div className="c-step-side-info-details">
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Name</div>
          <div className="b-step-side-info-field__value">{step.taskName}</div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Type</div>
          <div className="b-step-side-info-field__value">{"Task"}</div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Status</div>
          <div className="b-step-side-info-field__value">
            {ACTIVITY_STATUSES_TO_TEXT[step.flowTaskStatus]}
            <img src={ACTIVITY_STATUSES_TO_ICON[step.flowTaskStatus]} alt="Status" />
          </div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Start Time</div>
          <div className="b-step-side-info-field__value">{moment(step.startTime).format("YYYY-MM-DD h:mm A")}</div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Duration</div>
          <div className="b-step-side-info-field__value">
            {step.duration ? getHumanizedDuration(step.duration) : "-----"}
          </div>
        </div>
      </div>
    </div>
  );
};

StepSideInfo.propTypes = {
  step: PropTypes.object.isRequired
};

export default StepSideInfo;
