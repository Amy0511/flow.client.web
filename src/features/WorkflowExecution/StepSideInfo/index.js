import React from "react";
import PropTypes from "prop-types";
import momentDurationFormatSetup from "moment-duration-format";
import moment from "moment";
import "./styles.scss";

const StepSideInfo = ({ step }) => {
  momentDurationFormatSetup(moment);

  return (
    <div className="c-step-side-info">
      <div className="s-step-side-info-title">Summary</div>
      <div className="c-step-side-info-details">
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Name</div>
          <div className="b-step-side-info-field__value">{step.stepName}</div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Type</div>
          <div className="b-step-side-info-field__value">Task</div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Status</div>
          <div className="b-step-side-info-field__value">{step.flowTaskStatus}</div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Start Time</div>
          <div className="b-step-side-info-field__value">{moment(step.startTime).format("YYYY-MM-DD h:mm A")}</div>
        </div>
        <div className="b-step-side-info-field">
          <div className="b-step-side-info-field__key">Duration</div>
          <div className="b-step-side-info-field__value">{moment.duration(step.duration).format("hh:mm [mins]", { trim: false })}</div>
        </div>
      </div>
    </div>
  );
}

StepSideInfo.propTypes = {
  step: PropTypes.object.isRequired
};

export default StepSideInfo;
