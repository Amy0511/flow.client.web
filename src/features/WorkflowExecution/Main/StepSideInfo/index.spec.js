import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import StepSideInfo from "./index";

const step = {
  taskName: "test",
  flowTaskStatus: "completed",
  startTime: 1540912389131,
  duration: 2178
};

describe("StepSideInfo --- Snapshot", () => {
  it("Capturing Snapshot of StepSideInfo", () => {
    const renderedValue = renderer.create(<StepSideInfo step={step} />);
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("StepSideInfo --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<StepSideInfo step={step} />);
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
