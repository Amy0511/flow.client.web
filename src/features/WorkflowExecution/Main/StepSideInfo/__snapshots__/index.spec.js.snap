// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepSideInfo --- Snapshot Capturing Snapshot of StepSideInfo 1`] = `
<div
  className="c-step-side-info"
>
  <div
    className="s-step-side-info-title"
  >
    test
  </div>
  <div
    className="c-step-side-info-details"
  >
    <div
      className="b-step-side-info-field"
    >
      <div
        className="b-step-side-info-field__key"
      >
        Type
      </div>
      <div
        className="b-step-side-info-field__value"
      >
        Task
      </div>
    </div>
    <div
      className="b-step-side-info-field"
    >
      <div
        className="b-step-side-info-field__key"
      >
        Status
      </div>
      <div
        className="b-step-side-info-field__value"
      >
        Succeeded
        <img
          alt="Status"
          src="passed.svg"
        />
      </div>
    </div>
    <div
      className="b-step-side-info-field"
    >
      <div
        className="b-step-side-info-field__key"
      >
        Start Time
      </div>
      <div
        className="b-step-side-info-field__value"
      >
        October 30th 2018, 11:13:09 am
      </div>
    </div>
    <div
      className="b-step-side-info-field"
    >
      <div
        className="b-step-side-info-field__key"
      >
        Duration
      </div>
      <div
        className="b-step-side-info-field__value"
      >
        2 seconds
      </div>
    </div>
  </div>
</div>
`;
